graph/src/blockchain/block_stream.rs

```diff
@@ -274,17 +275,73 @@ pub trait FirehoseMapper<C: Blockchain>: Send + Sync {
     ) -> Result<BlockPtr, Error>;
 }
 
+#[async_trait]
+pub trait SubstreamsMapper<C: Blockchain>: Send + Sync {
+    async fn to_block_stream_event(
```

Comment:
this seems like the same as Mapper from the substreams crate? I think this is one for the next PR as well